libs/dyn/convert/normalize.go

```diff
@@ -225,9 +241,16 @@ func (n normalizeOptions) normalizeSlice(typ reflect.Type, src dyn.Value, seen [
 		return dyn.NewValue(out, src.Location()), diags
 	case dyn.KindNil:
 		return src, diags
-	}
 
-	return dyn.InvalidValue, diags.Append(typeMismatch(dyn.KindSequence, src, path))
+	default:
+		// Return verbatim if it's a pure variable reference.
+		if dynvar.IsPureVariableReference(src.MustString()) {
+			return src, nil
+		}
```

Comment:
This branch is not covered. If the input value is a type not captured in the switch/case, this will panic.

The `MustString` function panics if the type doesn't match the type you expect.

You could add a `dyn.KindString` branch and let it return if it is a pure reference and fall through otherwise.

Reply:
Fixed